GitUI/GitUICommands.cs

```diff
@@ -1487,6 +1487,12 @@ private bool RunCommandBasedOnArgument(IReadOnlyList<string> args, IReadOnlyDict
                         return StartCloneDialog(null, args[1].Replace("github-mac://openRepo/", ""), true);
                     }
 
+                    // User supplied a path. Open the repository if its a valid path
+                    if (Directory.Exists(args[1]))
+                    {
+                        return RunOpenRepoFromPath(args[1]);
```

Comment:
Suggest reducing the code (untested)
```suggestion
                        List<string> args2 = new() { "browse" };
                        args2.AddRange(args);
                        return RunBrowseCommand(args2);
```

Reply:
I am afraid `RunBrowseCommand` will not re-evaluate the path because it has already been checked and fed to the `Module`.

`RunOpenRepoFromPath` is less flexible than the existing logic in `Program.GetWorkingDir`, which should be re-used definitely.
I suggest to just call `LaunchBrowse(args[1])` - also if `File.Exists(args[1])`.
This has the drawback that a new instance is started. But  this should not be too expensive nowadays.

Reply:
We may not do this approach for reducing the code. Because `browse` or `openrepo` commands are trying to open a repository that is set during the instantiation (`GitModule`). 
So, instead of proceeding with a workaround, my opinion is to treat it as a separate function as now.

Please suggest if you have disagreements.

Reply:
> So, instead of proceeding with a workaround, my opinion is to treat it as a separate function as now.

I disagree to duplicating or - even worse - creating slightly different logic.

You can ignore the returned `IProcess` instance (as all other callers do, too). The new GE instance shall run detached.
Just `return true` in order to exit the first GE instance.

Reply:
mstv suggestion avoids the additional method
(I will not force checking if args[1] is a file to use the directory).
```suggestion
LaunchBrowse(args[1]);
```

Reply:
Sure. I have added those changes. I kept the Directory Exisist checks there, to not break the code and avoid the malformed commands. 

@mstv @gerhardol Please review.